src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiVariableConfigurationEditor.js

```diff
@@ -0,0 +1,169 @@
+import {FieldArray, useFormikContext} from 'formik';
+import isEqual from 'lodash/isEqual';
+import React, {useContext} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useAsync, useToggle} from 'react-use';
+
+import {APIContext} from 'components/admin/form_design/Context';
+import {REGISTRATION_OBJECTS_TARGET_PATHS} from 'components/admin/form_design/constants';
+import Field from 'components/admin/forms/Field';
+import Fieldset from 'components/admin/forms/Fieldset';
+import FormRow from 'components/admin/forms/FormRow';
+import {TextInput} from 'components/admin/forms/Inputs';
+import Select, {LOADING_OPTION} from 'components/admin/forms/Select';
+import {post} from 'utils/fetch';
+
+import {asJsonSchema} from './utils';
+
+/**
+ * Returns the Objects API Configuration editor modal for a specific variable. This only applies to V2 Options
+ *
+ * @typedef {{
+ *   version: 1 | 2;
+ *   objecttype: string;
+ *   objecttypeVersion: number;
+ *   variablesMapping: {variableKey: string, targetPath: string[]}[];
+ * }} ObjectsAPIRegistrationBackendOptions
+ *
+ * @param {Object} p
+ * @param {Object} p.variable - The current variable
+ * @returns {JSX.Element} - The summary, represented as a the parts of the target path separated by '>'
+ */
+const ObjectsApiVariableConfigurationEditor = ({variable}) => {
+  const {csrftoken} = useContext(APIContext);
+
+  const [jsonSchemaVisible, toggleJsonSchemaVisible] = useToggle(false);
+  const {values: backendOptions, getFieldProps, setFieldValue} = useFormikContext();
+
+  /** @type {ObjectsAPIRegistrationBackendOptions} */
+  const {objecttype, objecttypeVersion, variablesMapping, version} = backendOptions;
+  if (version !== 2) throw new Error('Not supported, must be config version 2.');
+
+  // get the index of our variable in the mapping, if it exists
+  let index = variablesMapping.findIndex(
+    mappedVariable => mappedVariable.variableKey === variable.key
+  );
+  let mappedVariable = variablesMapping[index];
+  if (index === -1) {
+    // if not found, grab the next available index to add it as a new record/entry
+    index = variablesMapping.length;
+    mappedVariable = {
+      variableKey: variable.key,
+      targetPath: undefined,
+    };
+    setFieldValue(`variablesMapping.${index}`, mappedVariable);
```

Comment:
you really need to get used to perform these sort of updates in an effect, not during rendering :grimacing: 

these formik callbacks do more than just updating state too, they trigger validation runs, so you can get in a render-cascade. The `useState` callback can be called during render, but only inside the same component as React somehow gives that special treatment (?)

Reply:
My understanding is that this code path won't be hit more than once, as the variablesMapping will then contain the newly added entry?

Reply:
And I also don't understand why there isn't a better api with functional components to initialize once other than using `useEffect` with empty dependencies..